tests/iterators_tests/test_balanced_serial_iterator.py

```diff
@@ -88,6 +115,14 @@ def _test_balanced_serial_iterator_no_batch_balancing():
     assert numpy.sum(labels_batch == 1) == 3
     assert numpy.sum(labels_batch == 2) == 3
 
+    iterator.serialize(DummySerializer())
+    assert iterator.current_position == 0
+    assert iterator.epoch == 1
+    assert iterator.is_new_epoch == True
+    for order in iterator._order:
+        assert order <= 9
+    assert iterator._previous_epoch_detail == 0.0
```

Comment:
You do not have to check the value of private properties as its specification is not documented. Ideally, we should write tests with documented APIs.